extensions/amp-selector/0.1/amp-selector.js

```diff
@@ -67,6 +71,44 @@ export class AmpSelector extends AMP.BaseElement {
     }
   }
 
+  /** @override */
+  mutatedAttributesCallback(mutations) {
+    mutations.forEach(mutation => {
+      const newValue = mutation.newValue;
+
+      switch (mutation.name) {
+        case 'selected':
+          if (!newValue) {
+            this.clearAllSelections_();
+            return;
+          }
+          // Generate map from comma-delimited option values in `newValue`.
+          const selectedArray = String(newValue).split(',');
+          const selectedMap = selectedArray.reduce((map, value) => {
+            map[value] = true;
+            return map;
+          }, {});
```

Comment:
also we could do selectedOptions = [selectedOptions[0]] , if you dont like splice for the multiple selection case and we could get rid of the break condition below.

IMO splice on such a small array is probably negligible and break is certainly confusing :)